app/src/main/java/com/example/resoluteapp/InboxFragment.java

```diff
@@ -40,56 +39,39 @@ public class InboxFragment extends Fragment {
 
     private FragmentInboxBinding binding;
 
-    FirebaseFirestore DB = FirebaseFirestore.getInstance();
+    private FirebaseFirestore DB = FirebaseFirestore.getInstance();
+
+    // Variable to store the username retrieved from SharedPreferences
+    // This variable holds the username of the current user, which is obtained from SharedPreferences.
+    // It is used to identify the user and perform user-specific operations throughout the fragment.
+    private String spUsername;
+
 
     @Override
-    public View onCreateView(
-            LayoutInflater inflater, ViewGroup container,
-            Bundle savedInstanceState
-    ) {
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         binding = FragmentInboxBinding.inflate(inflater, container, false);
+        spUsername = ((MainActivity) requireActivity()).getUsername();
 
         //calls the fillTable() function when the page is created so it is filled simultaneously
         try {
             fillTable();
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        } catch (ExecutionException e) {
-            throw new RuntimeException(e);
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
         }
-
         return binding.getRoot();
-
     }
 
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-
         //Friends Button
-        binding.toFriendsFromInbox.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                NavHostFragment.findNavController(InboxFragment.this)
-                        .navigate(R.id.action_inboxFragment_to_friendsFragment);
-            }
-        });
-
+        binding.toFriendsFromInbox.setOnClickListener(v -> NavHostFragment.findNavController(InboxFragment.this).navigate(R.id.action_inboxFragment_to_friendsFragment));
         //Home Button
-        binding.toHomeFromInbox.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                NavHostFragment.findNavController(InboxFragment.this)
-                        .navigate(R.id.action_inboxFragment_to_homeFragment);
-            }
-        });
-
+        binding.toHomeFromInbox.setOnClickListener(v -> NavHostFragment.findNavController(InboxFragment.this).navigate(R.id.action_inboxFragment_to_homeFragment));
```

Comment:
Why change the onClick() listeners for navigation? Unnecessary, and creates problems for #57.